test/e2e/web_console_plugin_test.go

```diff
@@ -0,0 +1,64 @@
+package e2e
+
+import (
+	"fmt"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+type webConsolePluginTest struct {
+	suite.Suite
+	wait.Awaitilities
+}
+
+func TestRunWebConsolePluginTest(t *testing.T) {
+	suite.Run(t, &webConsolePluginTest{})
+}
+
+func (s *webConsolePluginTest) SetupSuite() {
+	s.Awaitilities = WaitForDeployments(s.T())
+}
+
+func (s *webConsolePluginTest) TearDownTest() {
+	hostAwait := s.Host()
+	memberAwait := s.Member1()
+	memberAwait2 := s.Member2()
+	hostAwait.Clean(s.T())
+	memberAwait.Clean(s.T())
+	memberAwait2.Clean(s.T())
+}
+
+func (s *webConsolePluginTest) TestWebConsoleDeployedSuccessfully() {
+	// Provision a new user
+	hostAwait := s.Host()
+	hostAwait.UpdateToolchainConfig(s.T(), testconfig.AutomaticApproval().Enabled(true))
+
+	response := NewSignupRequest(s.Awaitilities).
+		Username("consoletest").
+		Email("consoletest@redhat.com").
+		EnsureMUR().
+		RequireConditions(ConditionSet(Default(), ApprovedAutomatically())...).
+		Execute(s.T()).GetSignupResponse(s.T())
+
+	// Response should contain a ConsoleURL with a value something like:
+	// https://console-openshift-console.apps.99b682869228f7464338-mgmt.ci.hypershift.devcluster.openshift.com/
+	require.Contains(s.T(), response, "consoleURL")
+
+	consoleURL := response["consoleURL"]
+	manifestURL := fmt.Sprintf("%s%s", consoleURL, "api/plugins/toolchain-member-web-console-plugin/plugin-manifest.json")
+
+	resp, err := http.Get(manifestURL)
```

Comment:
I think before calling the service endpoint we should wait for the health endpoint returning 200 to make sure the service is fully deployed. Just to avoid any test flakiness.

Reply:
Have added a check that will retry a few times if the health check endpoint fails.